test/lib/writers/RulesWriter.test.ts

```diff
@@ -0,0 +1,37 @@
+import { OutputFormat } from '@salesforce/code-analyzer-core';
+import fs from 'node:fs';
+import path from 'node:path';
+import { RulesFileWriter } from "../../../src/lib/writers/RulesWriter";
+import * as Stub from '../../stubs/StubRuleSelection';
+
+describe('RulesWriter', () => {
+
+    let writeFileSpy: jest.SpyInstance;
+    let writeFileInvocations: { file: fs.PathOrFileDescriptor, contents: string | ArrayBufferView }[];
+
+    beforeEach(() => {
+        writeFileInvocations = [];
+        writeFileSpy = jest.spyOn(fs, 'writeFileSync').mockImplementation((file, contents) => {
+            writeFileInvocations.push({file, contents});
+        });
+    });
+
+    it('Rejects invalid file format', () => {
+        const invalidFile = 'file.xml';
+        expect(() => new RulesFileWriter(invalidFile)).toThrow(invalidFile);
+    });
+
+    it('Writes to a json file path', () => {
+        const outfilePath = path.join('the', 'results', 'path', 'file.json');
+        const expectations = {
+            file: outfilePath,
+            contents: `Rules formatted as ${OutputFormat.JSON}`
+        };
+        const rulesWriter = new RulesFileWriter(expectations.file);
+        const stubbedEmptyRuleSelection = new Stub.StubEmptyRuleSelection();
+        rulesWriter.write(stubbedEmptyRuleSelection);
```

Comment:
I thought about testing "more" rules, but this class shouldn't care about the number of rules, since it's just getting back whatever the formatted output string is anyways. I could add an extra test here to handle an error condition or an empty string if that would be good coverage though.

Reply:
IMO, there should probably be a test case for zero rules and a test case for one rule, since those are the boundary cases.